.github/workflows/full-unittests.yml

```diff
@@ -0,0 +1,107 @@
+name: cellxgene_census package full unit tests
+
+# Run all unit tests, including those that are too expensive to run frequently.
+# This workflow requires a very large capacity runner, e.g., 1+TiB RAM, which is
+# currently available through self-hosted runners. These runners have no swap,
+# so an OOM will cause the workflow to fail with OOMKilled (exit code 137)
+
+on:
+  schedule:
+    - cron: "0 1 * * 6" # every Saturday night, 1AM UTC
+
+  workflow_dispatch: # used for debugging or manual validation of a branch
+    inputs:
+      tiledbsoma_python_dependency:
+        # Accepts any package spec that pip understand, e.g.,
+        #   tiledbsoma==1.0
+        #   git+https://github.com/single-cell-data/TileDB-SOMA.git#egg=tiledbsoma&subdirectory=apis/python/
+        #   git+https://github.com/single-cell-data/TileDB-SOMA.git@release-1.6#egg=tiledbsoma&subdirectory=apis/python/
+        # or whatever...
+        description: "tiledbsoma package specified as pip requirement"
+        required: false
+        default: ""
+        type: string
+
+jobs:
+  py_unit_tests:
+    runs-on: single-cell-1tb-runner
+    strategy:
+      fail-fast: false # prevent this job from killing other jobs
+    steps:
+      - name: log system state
+        run: |
+          free
+          echo ---------
+          df -kh
+          echo ---------
+          lscpu
+
+      - name: install OS dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y build-essential git-all libxml2-dev libssl-dev libcurl4-openssl-dev cmake
+
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: install python dependencies (including experimental)
+        run: |
+          python -m pip install -U pip setuptools setuptools_scm wheel
+          pip install --use-pep517 accumulation-tree # Geneformer dependency needs --use-pep517 for Cython
+          pip install -r ./api/python/cellxgene_census/scripts/requirements-dev.txt
+          pip install './api/python/cellxgene_census/[experimental]'
+
+      - name: install tiledbsoma version override
+        if: github.event_name == 'workflow_dispatch' && github.event.inputs.tiledbsoma_python_dependency != ''
+        run: |
+          pip install -e '${{ github.event.inputs.tiledbsoma_python_dependency }}'
+
+      - name: pytest (--expensive --experimental)
+        run: |
+          echo '--------- tiledbsoma.show_package_version():'
+          python -c 'import tiledbsoma; tiledbsoma.show_package_versions()'
+          echo '--------- PIP package versions:'
+          pip freeze
+
+          PYTHONPATH=. pytest -v --durations=0 -rP --experimental --expensive ./api/python/cellxgene_census/tests/
+
+  r_unit_tests:
+    if: false # TEMPORARILY DISABLE FOR DEBUGGING
+    runs-on: single-cell-1tb-runner
```

Comment:
woot! just wanted to say 👋🏻 and introduce myself. I worked with @beroy getting these self-hosted runners set up. I'm watching them periodically and recording metrics on the nodes and autoscaler, but ping me if you see any unusual behavior or have other requirements you want us to add to them. Have fun!

Reply:
hi @jakeyheath - these are a game changer.  Really happy to have them!

So far they seem to work great, other than missing some "nice-to-have" packages installed in their default image.  I have given @beroy an initial list.

Reply:
Easy to solve, we already chatted this morning about building a custom container for this project to use that has all the packages pre-installed. 

Let's meet in the new year and see how things are going. Right now, everything is running in our czi-playground account. Might be good to put this project is a more specific location and tune the performance as needed. Here's a screenshot of the metrics I collected on the latest run. Happy to show yall how to pull these yourselves and tune to your needs.


<img width="1576" alt="Screenshot 2023-12-18 at 12 15 17 PM" src="https://github.com/chanzuckerberg/cellxgene-census/assets/76011913/22be331e-48e8-4593-933e-4ecd9ea30977">